lib/install/management/store_files.go

```diff
@@ -53,15 +53,37 @@ func (d *Dispatcher) DeleteStores(vchVM *vm.VirtualMachine, conf *metadata.Virtu
 	var emptyImages bool
 	var emptyVolumes bool
 	log.Infof("Removing images")
-	if emptyImages, err = d.deleteImages(ds, p); err != nil {
-		errs = append(errs, err.Error())
+	for _, imageDir := range conf.ImageStores {
+		imageDSes, err := d.session.Finder.DatastoreList(d.ctx, imageDir.Host)
```

Comment:
@hickeng I actually wrote (most) of those functions that Faiyaz started -- I haven't changed how datastore URLs are parsed in vic-machine, although I do agree that those should be updated to use generic parsing functions (if not the ones that I wrote, another one that uses the same regexes to go from `ds://host/path` format to `url.URL` because that function doesn't exist currently) but if that needs to be done I'd like to do it separately from this PR since as it stands the parsing is unchanged at the moment.

At the point in the code where you're commenting, the URL is already parsed into a proper url.URL object.